src/protocols/TRXLooper.cpp

```diff
@@ -623,83 +254,40 @@ void TRXLooper::Stop()
     mStreamEnabled = false;
 }
 
-bool TRXLooper::IsStreamRunning()
-{
-    return mStreamEnabled;
-}
-
-int TRXLooper::StreamRx(lime::complex32f_t* const* dest, uint32_t count, SDRDevice::StreamMeta* meta)
+template<class T> uint32_t TRXLooper::StreamRxTemplate(T* const* dest, uint32_t count, SDRDevice::StreamMeta* meta)
 {
     bool timestampSet = false;
     uint32_t samplesProduced = 0;
     const bool useChannelB = mConfig.channels.at(TRXDir::Rx).size() > 1;
 
-    lime::complex32f_t* f32_dest[2] = { static_cast<lime::complex32f_t*>(dest[0]),
-        useChannelB ? static_cast<lime::complex32f_t*>(dest[1]) : nullptr };
-
     bool firstIteration = true;
 
     //auto start = high_resolution_clock::now();
     while (samplesProduced < count)
     {
-        if (!mRx.stagingPacket && !mRx.fifo->pop(&mRx.stagingPacket, firstIteration, 2000))
+        if (!mRx.stagingPacket && !mRx.fifo->pop(&mRx.stagingPacket, firstIteration, 250))
```

Comment:
timeout has to be 2000ms. If stream start is configured to wait for GPS PPS signal, reading data might timeout before it actually started.